kvmagent/kvmagent/plugins/hygon_device_plugin.py

```
@@ -0,0 +1,538 @@
+
+        Returns:
+            list: List of CCP device dictionaries with keys:
+                - pciBdf (str): PCI address (e.g., "0000:06:00.1")
+                - deviceType (str): Device type with quote (e.g., "'NTBCCP")
+                - deviceId (str): Device ID with quote (e.g., "1468'")
+                - driverStatus (str): Current driver binding (e.g., "drv=hct")
+
+        Raises:
+            Exception: If failed to execute hct_ccp_bind.py or parse output
+        """
+        r, o, e = bash_roe("python %s -s" % self.HCT_CCP_BIND_SCRIPT)
```

Ruff (0.14.7)

`bash_roe` may be undefined, or defined from star imports

(F405)